core/lib/src/config/secret_key.rs

```diff
@@ -178,6 +187,75 @@ impl SecretKey {
     {
         ser.serialize_bytes(&[0; 32][..])
     }
+
+    /// Encrypts the given plaintext.
+    /// Generates a random nonce for each encryption to ensure uniqueness.
+    /// Returns the base64-encoded string of the concatenated nonce and ciphertext.
+    ///
+    /// # Example
+    /// ```rust
+    /// let plaintext = "I like turtles";
+    /// let secret_key = SecretKey::generate().expect("error generate key");
+    ///
+    /// let encrypted = secret_key.encrypt(&plaintext).expect("can't encrypt");
+    /// let decrypted = secret_key.decrypt(&encrypted).expect("can't decrypt");
+    ///
+    /// assert_eq!(decrypted, plaintext);
+    /// ```
+    pub fn encrypt(&self, plaintext: &str) -> Result<String, &'static str> {
+        // Convert the encryption key to a fixed-length array
+        let key: [u8; KEY_LEN] = self.key.encryption().try_into().map_err(|_| "enc key len error")?;
+
+        // Create a new AES-256-GCM instance with the provided key
+        let aead = Aes256Gcm::new(GenericArray::from_slice(&key));
+
+        // Generate a random nonce
+        let mut nonce = [0u8; NONCE_LEN];
+        let mut rng = rand::thread_rng();
+        rng.try_fill_bytes(&mut nonce).map_err(|_| "couldn't random fill nonce")?;
+        let nonce = Nonce::from_slice(&nonce);
+
+        // Encrypt the plaintext using the nonce
+        let ciphertext = aead.encrypt(nonce, plaintext.as_ref()).map_err(|_| "encryption error")?;
+
+        // Prepare a vector to hold the nonce and ciphertext
+        let mut encrypted_data = Vec::with_capacity(NONCE_LEN + ciphertext.len());
+        encrypted_data.extend_from_slice(nonce);
+        encrypted_data.extend_from_slice(&ciphertext);
+
+        // Return the base64-encoded result
+        Ok(URL_SAFE.encode(encrypted_data))
```

Comment:
Should we enforce Base64 encoding? I think a more general solution might be to simply return `Vec<u8>`, and let the user encode it if they want to.

Reply:
Good point!
In this case we don't have to add `base64` lib to Rocket and this option is more flexible. 
I will change code and example.